src/core/border_router/routing_manager.cpp

```diff
@@ -3601,6 +3600,11 @@ void RoutingManager::PdPrefixManager::EvaluateStateChange(Dhcp6PdState aOldState
     case kDhcp6PdStateRunning:
         break;
     }
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
+#if !OPENTHREAD_BORDER_ROUTING_DHCP6_PD_REQUESTOR
```

Comment:
I guess `OPENTHREAD_BORDER_ROUTING_DHCP6_PD_REQUESTOR` is a new config?

Let's use `OPENTHREAD_CONFIG_...` prefix for it?
Also let's define it in the `config/border_routing.h` header (for its default value) and document its function.